src/checkers/firefoxchecker.py

```
@@ -0,0 +1,162 @@
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+import functools
+import json
+import logging
+import os
+import re
+import urllib.request
+
+from lib.externaldata import ExternalData, ExternalDataSource, Checker
+from lib import utils
+
+log = logging.getLogger(__name__)
+
+
+class FirefoxChecker(Checker):
+    FIREFOX_RELEASE_INFO_URL = 'https://product-details.mozilla.org/1.0/firefox_versions.json'
+    FIREFOX_ARCHIVE_BASE_URL = 'https://archive.mozilla.org/pub/firefox/releases/{version}/'
+    FIREFOX_PLATFORM = 'linux-x86_64'
+    CHECKER_BROWSER_SOURCE_FILENAME = 'firefox.tar.bz2'
+    CHECKER_BROWSER_FILENAME_MATCH = r'firefox-.+\.tar\.bz2$'
+    CHECKER_BROWSER_DEFAULT_LANGUAGE = 'en-US'
+    CHECKER_TRANSLATIONS_FILENAME_MATCH = r'.+\.xpi$'
+
+    def _should_check(self, module_data):
+        return module_data.checker_data.get('type') == 'firefox'
+
+    def check_module(self, module_data, external_data):
+        if not self._should_check(module_data):
+            return
+
+        log.debug('Retrieving latest available firefox version')
+        # This should raise an error if either the release info url or
+        # the returned json data is broken. Lets not catch it here as
+        # we want that checker to break if that happens.
+        latest_version = self._get_latest_available_version()
+        assert latest_version != None
+
+        log.debug('Latest available firefox version: %s', latest_version)
+
+        log.debug('Retrieving latest firefox version info')
+        latest_info = None
+        try:
+            latest_info = self._get_latest_version_info(latest_version)
+        except Exception as e:
+            log.warning('Error retrieving latest firefox version info: %s', e)
+        finally:
+            if not latest_info:
+                log.warning('Could not retrieve latest firefox version information')
```

Again I would prefer to skip the exception handling and just let it bubble up in this case. We won't notice the warnings, but we will notice the job failing.